apis/nucleus/src/index.js

```diff
@@ -141,6 +141,12 @@ const DEFAULT_CONTEXT = /** @lends Context */ {
   keyboardNavigation: false,
   /** @type {boolean=} */
   disableCellPadding: false,
+  /**
+   * Type used for toggling to the data view (toggleDataView)
+   * This type need to be registered as well
+   * @type {string=}
+   * */
+  dataViewType: 'sn-table',
```

Comment:
One thing we could do to help this out is to check for types when doing the toggling. There are basically two options for registering your table chart, either "sn-table" or "table". The "table" one should be quite common as it allows for embedding the old table as the new.
Could be when running toggle explicitly check for the registred types and then use one of the two if either is registred and this "dataViewType" isn't set? (so default it to an empty string, meaing somewhat "auto")

Reply:
Could look something like this https://github.com/qlik-oss/nebula.js/pull/1418
I put that logic in types on halo since it had easier access to the type list. Otherwise you would need to check them one by one inside the toggleDataView logic

Reply:
yea, thats pretty nice. Could actually keep the default value of dataViewType and just use the others as a fallback instead.
I'd just argue for not adding a getDataViewType function to types and instead just either use the getSupportedVersion instead and keep the check in toggleDataView or add a function to get the object map as opposed to an array from getList.

Reply:
Updated it now. Keeping `sn-table` as default and checking if the dataViewType is supported. If not there is the fallback of using `table` as the type.  The nice thing with this is that we only need one fallback. If you change the default dataViewType I think it can also be expected that you have registered that type